rootex/framework/components/space/transform_component.cpp

```diff
@@ -24,8 +24,14 @@ TransformComponent::TransformComponent(Entity& owner, const JSON::json& data)
 
 void TransformComponent::updateAbsoluteTransformValues()
 {
+	Vector3 scale = m_AbsoluteScale;
+	Vector3 pos = m_AbsolutePosition;
+	Quaternion rot = m_AbsoluteRotation;
 	m_AbsoluteTransform = m_TransformBuffer.transform * m_ParentAbsoluteTransform;
+	m_TransformBuffer.transform.Decompose(scale, rot, pos);
```

Comment:
Why do we need changes in this function? This function is run fairly often during the frame and it was already doing all it needs to do. The extra computations here don't actually do anything here.